jdaviz/components/viewer_data_select.vue

```diff
@@ -194,6 +196,9 @@ module.exports = {
       return this.$props.data_items.filter((item) => this.itemIsVisible(item, false))
     },
     extraDataItems() {
+      if (this.$props.viewer.config === 'mosviz') {
+        return []
+      }
```

Comment:
I'm not sure this will be sufficient... what happens if you create a model and unload it from the viewer (but its still assigned to the active row)?  Does it show in the "extra items" dropdown or does it disappear forever?

Reply:
Aha! You've run into the second part of that commit! https://github.com/spacetelescope/jdaviz/pull/1790/commits/668a2bb1b65e441c5076ef4229dfe71f51ee01b0#diff-671bb80f9bf0db3b85cab9b0e24692dc85215aa08b106a50c148914e8d7a8e0fR145-R146

This is a second improvement I ran past @camipacifici; I'm basically hacking the functionality of the bug found fixed in #1808 / #1798 so that all data from the same row is always at the top (never in the dropdown). Therefore, your model, and all datasets assigned to the same row, will always be there, even though I removed the dropdown for the "other" datasets

Reply:
Might be controversial, but figured I'd put it here and see what people think!

Reply:
This now behaves differently than all the other configs, where items that are valid to load, but not currently loaded into the viewer, fall below the expanded menu.

<img width="576" alt="image" src="https://user-images.githubusercontent.com/877591/199734430-1114aea4-b152-40c2-8d73-05ec8c24f11e.png">

I think we still want the expanded menu in this case (although the text that mentions "incl other rows" will need to be removed).

Reply:
Considering that the behavior common to the other configs is very annoying for Mosviz, I think there is no harm in having something different here. We can give it a try and revert or move to a new solution when a better approach is found.

Reply:
Couple thoughts on the above:

1. Sorting would definitely be possible, though it would probably come at the cost of performance; we might need to loop through all the data entries every time before being able to sort, since i don't think we control the order the entries come in through the loop
2. We could also change it so that you can't "unload" the data of the same row, and can only toggle the visibility?

Reply:
Sorry - I must be missing some context.  What is the annoying case we're trying to improve?  I just remember a lot of discussions with @Jenneh when designing the menu and if I remember correctly the idea of the expanded menu was to avoid user confusion about the difference between loading/unloading vs toggling visibility, and this entirely reverts that and only for mosviz.  If the icons/tooltips are sufficient to make that distinction, then I think we should remove the expanded menu across all tools.

Reply:
The context here is if you generate data (say with gaussian smooth) and unload it, that spectrum gets dropped all the way to the bottom of the dropdown, requiring you to sift through all of the 1D spectra to find the data relevant to the current row.

https://user-images.githubusercontent.com/25206008/199738792-9e7523fb-5c99-4677-822e-18ce96bac134.mp4

This is the behavior Cami is describing as very annoying

Reply:
Yes, but doesn't this PR remove all the other entries from other rows?  I agree this is an annoyance we should address if/when we re-introduce those rows, but would no longer be an issue with this PR in place, right?

Reply:
The removal of the 1D links is intended to be a temporary solution. We intend to bring it back, hence why I'm thinking of this change in the context of the eventual goal of the 1D spectra returning. The solution we implement here should be compatible with the eventuality that the 1D spectra will come back.

(So a beneficial byproduct of the change here is if we don't put the row-relevant data in the dropdown, the solution here is as simple as the code you highlighted above. If not, then we'd have to add some additional filtering logic as you described.)

Reply:
> If not, then we'd have to add some additional filtering logic

Not to say that this is even difficult. Though if the simpler solution also solves this annoyance, then I felt it was at least worth it to consider (hence why we're here 😄 )

Reply:
Ok, can we attempt a different solution with sorting that isn't inconsistent with the other tools and see if there are actually performance issues (that is definitely a valid point)?

I could either spend a few hours playing with that outside of this PR (with all rows) to see if its feasible before we make the decision here, or we could defer until the rows are reintroduced (in which case I would personally prefer just removing the rows but keeping the expanding menu for now and keeping this solution in our back pocket as one option when we get to that rather than possibly switching back and forth).

Reply:
Sorry, I missed the information that rows are not to be reintroduced yet. I guess I will let you figure this out because my opinion is too much outside the code and we all agree what the end product should be. Thank you

Reply:
Based on the team discussion, this change seems to require more thought. I'll change the filtering logic to preserve the "other" dropdown and we'll kick this can down the road. Thanks for the input, team!